src/features/SolarSystem/SolarRender.cpp

```diff
@@ -0,0 +1,297 @@
+#pragma once
+
+#include <Renderer.h>
+#include <gl\GL.h>
+#include <gl\GLU.h>
+/* file for SOIL library to load the image*/
+#include "..\tools\soil\src\SOIL.h"
+#include"features\SolarSystem\SolarRender.h"
+using namespace Features;
+
+void SolarSystemRenderer::update()
+{
+	g_Angle = g_Angle + 0.1f;
+	if (g_Angle >= 360)
+		g_Angle = 0.1f;
+	g_Mercury = g_Mercury + 0.7f;
+	g_Venus = g_Venus + 0.3f;
+	g_Earth = g_Earth + 0.5f;
+	g_Mars = g_Mars + 0.3f;
+	g_Jupiter = g_Jupiter + 0.5f;
+	g_Saturn = g_Saturn + 0.6f;
+	g_Uranus = g_Uranus + 0.7f;
+	g_Neptune = g_Neptune + 0.5f;
+}
+int SolarSystemRenderer::LoadTexture(GLuint *texture, char *szPath)
+{
+	int 	iTextureWidth, iTextureHeight;
+	unsigned char 	*pImageData = NULL;
+	int iStatus = FALSE;
+	pImageData = SOIL_load_image((const char*)szPath, &iTextureWidth, &iTextureHeight, 0, SOIL_LOAD_RGB);
+	if (pImageData!=NULL)
+	{
+
+		iStatus = TRUE;
+		glGenTextures(1, texture);
+		glBindTexture(GL_TEXTURE_2D, *texture);
+		glPixelStorei(GL_UNPACK_ALIGNMENT, 4);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+		gluBuild2DMipmaps(GL_TEXTURE_2D, 3, iTextureWidth, iTextureHeight, GL_RGB, GL_UNSIGNED_BYTE, (void *)pImageData);
+		SOIL_free_image_data(pImageData); // free the imageData
+	}
+	return(iStatus);
+}
+
+const char* SolarSystemRenderer::GetName()
+{
+	static const char name[] = "SolarSystemRenderer";
+	return name;
+}
+
+RendererResult SolarSystemRenderer::Initialize(Window window)
+{
+	glEnable(GL_TEXTURE_2D);
+	g_Mercury = 0.0f;
+	g_Venus = 0.0f;
+	g_Earth = 0.0f;
+	g_Mars = 0.0f;
+	g_Jupiter = 0.0f;
+	g_Saturn = 0.0f;
+	g_Uranus = 0.0f;
+	g_Neptune = 0.0f;
+	g_Angle = 0.0f;
+	LoadTexture(&gbTexture[0], "\\git\\project\\src\\features\\SolarSystem\\earth.bmp");
```

Comment:
Looks to be hardcoded path so might not work for everybody. 

Let me tune the code a bit to be able to locate these resources.